src/main/java/sirius/biz/storage/layer3/VirtualFile.java

```diff
@@ -0,0 +1,794 @@
+                return canProvideOutputStream.test(this);
+            }
+
+            return outputStreamSupplier != null;
+        } catch (Exception e) {
+            throw handleErrorInCallback(e, "canProvideOutputStream");
+        }
+    }
+
+    /**
+     * Tries to create an output stream to write to the file.
+     *
+     * @return an output stream to provide the contents of the child or an empty optional if no output stream can be created
+     */
+    public Optional<OutputStream> tryCreateOutputStream() {
+        try {
+            if (outputStreamSupplier == null) {
+                return Optional.empty();
+            }
+
+            return Optional.ofNullable(outputStreamSupplier.apply(this));
+        } catch (Exception e) {
+            throw handleErrorInCallback(e, "outputStreamSupplier");
+        }
+    }
+
+    /**
+     * Creates an output stream to write to the file.
+     *
+     * @return an output stream to provide the contents of the child
+     * @throws HandledException if no output stream could be created
+     */
+    public OutputStream createOutputStream() {
+        return tryCreateOutputStream().orElseThrow(() -> Exceptions.createHandled()
+                                                                   .withNLSKey("VirtualFile.cannotWrite")
+                                                                   .set("file", path())
+                                                                   .handle());
+    }
+
+    /**
+     * Determines if the file is readable.
+     *
+     * @return <tt>true</tt> if the file is readable, <tt>false</tt> otherwise
+     */
+    public boolean isReadable() {
+        return !isDirectory() && canCreateInputStream();
+    }
+
+    /**
+     * Determines if the file is writeable.
+     *
+     * @return <tt>true</tt> if the file is writeable, <tt>false</tt> otherwise
+     */
+    public boolean isWriteable() {
+        return !isDirectory() && canCreateOutputStream();
+    }
+
+    /**
+     * Determines if an {@link InputStream} can (probably) be created for this file.
+     * <p>
+     * Note that {@link #isReadable()} is probably the better check.
+     *
+     * @return <tt>true</tt> if an output stream can be created, <tt>false</tt> otherwise
+     */
+    public boolean canCreateInputStream() {
+        try {
+            if (canProvideInputStream != null) {
+                return canProvideInputStream.test(this);
+            }
+
+            return inputStreamSupplier != null;
+        } catch (Exception e) {
+            throw handleErrorInCallback(e, "canProvideInputStream");
+        }
+    }
+
+    /**
+     * Tries to create an input stream to read the contents of the file.
+     *
+     * @return an input stream to read the contents of the child or an empty optional if no input stream can be created
+     */
+    public Optional<InputStream> tryCreateInputStream() {
+        try {
+            if (inputStreamSupplier == null) {
```

Comment:
I think inverting this if check would improve readability/flow and would be consistent to the other methods in this class